### PR TITLE
Convert Status.Ready to plain bool

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by detiber** [comment]
lgtm, but want to give @ncdc a chance to look over.

File: pkg/apis/infrastructure/v1alpha2/awsmachine_types.go [hunk 1]
```diff
@@ -83,7 +83,8 @@ type AWSMachineSpec struct {
 // AWSMachineStatus defines the observed state of AWSMachine
 type AWSMachineStatus struct {
 	// Ready is true when the provider resource is ready.
-	Ready *bool `json:"ready,omitempty"`
+	// +optional
```
Comment (ncdc) [inline pkg/apis/infrastructure/v1alpha2/awsmachine_types.go]: I don't think you can call this optional unless it's a pointer - remove this?

Comment (vincepri) [reply pkg/apis/infrastructure/v1alpha2/awsmachine_types.go]: It seems controller tools allows it, if it’s not marked as optional, it’ll be marked as required in the CRD

Comment (ncdc) [reply pkg/apis/infrastructure/v1alpha2/awsmachine_types.go]: Does that mean it will error if it's not in the json/yaml when you POST it?

Comment (vincepri) [reply pkg/apis/infrastructure/v1alpha2/awsmachine_types.go]: Yes, the validation will fail, similar to what we've seen with ObjectMeta

Comment (ncdc) [reply pkg/apis/infrastructure/v1alpha2/awsmachine_types.go]: k, thanks.

**Pull request merged**